presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotPushdownUtils.java

```diff
@@ -163,6 +185,52 @@ public ExpressionType getExpressionType()
         return nodeBuilder.build();
     }
 
+    /**
+     * Try to push down query like: `SELECT count(distinct $COLUMN) FROM myTable` to Pinot as `SELECT distinctCount($COLUMN) FROM myTable`.
+     * This function only handles the case of an AggregationNode (COUNT on $COLUMN) on top of an AggregationNode(of non-aggregate on $COLUMN).
+     *
+     * @param nodeBuilder
+     * @param aggregationNode
+     * @param outputColumn
+     * @param aggregation
+     * @return true if push down successfully otherwise false.
+     */
+    private static boolean handlePushDownSingleDistinctCount(ImmutableList.Builder<AggregationColumnNode> nodeBuilder, AggregationNode aggregationNode, VariableReferenceExpression outputColumn, AggregationNode.Aggregation aggregation)
+    {
+        if (aggregation.getCall().getDisplayName().equalsIgnoreCase(COUNT_FUNCTION_NAME)) {
```

Comment:
Can we return early in this function for most of the `if`s; so we don't have to deal with all these braces
```java
if (!aggregation.getCall().getDisplayName().equalsIgnoreCase(COUNT_FUNCTION_NAME)) {
    return false;
}

...

Reply:
done.